### PR TITLE
Fix location path on web

### PR DESCRIPTION
**Pull request by nqhhdev**
### Issue related to wrong location when listening to change from auth

https://github.com/user-attachments/assets/c9b49668-85f1-4909-9fe3-f1776ddbd6a9

### Resolved

https://github.com/user-attachments/assets/339ff24c-932e-47cc-a44c-b2a47f88d566

- Stg env

https://github.com/user-attachments/assets/062991f9-d6ae-4e0c-bdb2-6e54cfe5351b

- Dev env

https://github.com/user-attachments/assets/efa5c14c-6d9a-4c8d-847a-10f2d09d3db2

- Linagora env

https://github.com/user-attachments/assets/be8bc693-1138-47ff-b5bc-7c16a4424786

- Matrix env

https://github.com/user-attachments/assets/0c9a324f-2e63-401e-9a7a-d157878c2375

### CODE REVIEW

**Comment by hoangdat** [comment]
still not work when run with configuration:
```
{
  "app_grid_dashboard_available": true,
  "application_name": "Twake Chat",
  "application_welcome_message": "Welcome to Twake Chat!",
  "default_homeserver": "matrix.linagora.com",
  "hide_redacted_events": false,
  "hide_unknown_events": false,
  "homeserver": "https://matrix.linagora.com/",
  "issue_id": "",
  "privacy_url": "https://twake.app/en/privacy/",
  "render_html": true
}
```

File: lib/presentation/mixins/connect_page_mixin.dart [hunk 1]
```diff
@@ -295,6 +295,15 @@ mixin ConnectPageMixin {
...
+    html.window.history.replaceState(
+      {},
+      '',
+      '${getBaseUrlBeforeHash(html.window.location.href)}#/${route ?? 'rooms'}',
+    );
+  }
+
+  String getBaseUrlBeforeHash(String fullUrl) {
+    final fragmentIndex = fullUrl.indexOf('#/');
+    return fragmentIndex != -1 ? fullUrl.substring(0, fragmentIndex) : fullUrl;
```
Comment (hoangdat) [inline lib/presentation/mixins/connect_page_mixin.dart:307]: write unit test for this function

File: test/utils/string_extension_test.dart [hunk 2]
```diff
@@ -598,5 +598,41 @@ void main() {
...
+    });
+
+    test('getBaseUrlBeforeHash handles URL with query parameters and hash', () {
+      const url = 'https://example.com/test?query=1#/section';
+      const expectedUrl = 'https://example.com/test?query=1';
+
+      final result = url.getBaseUrlBeforeHash();
+
+      expect(result, equals(expectedUrl));
+    });
```
Comment (hoangdat) [inline test/utils/string_extension_test.dart:636]: group it in one group test

Comment (nqhhdev) [reply test/utils/string_extension_test.dart:638]: Fixed

**Review by hoangdat — APPROVED** [review]

**Pull request merged**